judgels-backends/sandalphon/sandalphon-api/src/main/java/judgels/sandalphon/api/submission/bundle/ItemSubmission.java

```diff
@@ -15,6 +16,20 @@
     String getAnswer();
     String getUserJid();
     Instant getTime();
+    Optional<Grading> getGrading();
+
+    static ItemSubmission withoutGrading(ItemSubmission submission) {
+        return new ItemSubmission.Builder().from(submission).grading(Optional.empty()).build();
+    }
+
+    @Value.Immutable
+    @JsonDeserialize(as = ImmutableGrading.class)
+    interface Grading {
```

Comment:
Why not make this as a top-level interface `ItemGrading`?

Reply:
OK, changed